### PR TITLE
provisioner/local-exec: Allow passing env vars to commands

### LINKED ISSUE #13860: Feature request: allow defining environment variables for local-exec provisioner
It would be beneficial to be able to set environment variables that should be defined in the context that a `local-exec` provisioner is executed in, e.g.

```hcl
provisioner "local-exec" {
  env {
    NAME = "John"
  }

  command = "echo $NAME"
}
```

A lot of programs accept parameters in the form of environment variables.

### PR DESCRIPTION
**Pull request by cnicolov**
Fixes #13860

### CODE REVIEW

File: website/source/docs/provisioners/local-exec.html.markdown [hunk 1]
```diff
@@ -38,3 +52,6 @@ The following arguments are supported:
   as a relative path to the current working directory or as an absolute path.
   It is evaluated in a shell, and can use environment variables or Terraform
   variables.
+
+* `environment` - (Optional) Block of key value pairs reperesing the
```
Comment (dasch) [inline website/source/docs/provisioners/local-exec.html.markdown]: Typo: `s/reperesing/representing/`

**Comment by dasch** [comment]
Any update on this?

**Comment by dasch** [comment]
_bump_

**Comment by cnoffsin** [comment]
Please approve this.  This is especially needed when running ansible playbooks you have to specify the path to the roles and you need ansible_roles_path variable set.

**Comment by apparentlymart** [comment]
Hi all! Sorry for the long silence here. This got lost in the backlog, it seems.

Unfortunately this changeset came into conflict with another that added the `interpreter` attribute. This change seems fine to me otherwise, so @cnicolov if you are able to rebase this and resolve the conflicts I see no reason not to merge this. If you're not able to do further work here then no worries -- I know we let it sit for a long time -- just let me know and I can resolve the conflicts here myself when I have some moments.

**Comment by cnicolov** [comment]
@apparentlymart It's good to go now - just changed the test case a little bit (to match the upstream approach)

**Comment by cnicolov** [comment]
Hi @apparentlymart , I've resolved the conflicts again, this time for the `working_dir` attribute.

Could you please review and/or merge/close this PR? It's almost one year old 🎂

**Comment by apparentlymart** [comment]
Thanks for updating it again, @cnicolov, and sorry for repeatedly losing this off my radar. :confounded:

**Comment by kalyanrajsista** [comment]
I have the same problem here now. I want to set environment variables and use them in powershell script. Is the feature merge to use now?

**Pull request merged**